elastic4s-core/src/main/scala/com/sksamuel/elastic4s/searches/QueryApi.scala

```diff
@@ -258,8 +258,11 @@ trait QueryApi {
   def termsLookupQuery(field: String, path: String, ref: DocumentRef) =
     TermsLookupQuery(field, TermsLookup(ref, path))
 
-  def termsSetQuery(field: String, terms: Set[Any]): com.sksamuel.elastic4s.searches.queries.term.TermsSetQuery =
-    com.sksamuel.elastic4s.searches.queries.term.TermsSetQuery(field, terms)
+  def termsSetQuery(field: String, 
```

Comment:
Since one of the two minimumShouldMatch params have to be specified, I think we should have two methods here, one that takes a string, and one that takes a script. Then we can do,

`termsSetQuery(field, terms, min)` or `termsSetQuery(field, terms, script("script"))` rather than `termsSetQuery(field, terms, Some(min))`. If that's ok, please make the change and I'll merge.

Great pr thanks.

Reply:
True! Might be clearer when using the method. I'll make the change, thanks!